packages/jest-haste-map/src/index.js

```diff
@@ -325,6 +325,28 @@ class HasteMap extends EventEmitter {
           throw new Error(message);
         }
         this._console.warn(message);
+        // We do NOT want consumers to use a module that is ambiguous.
+        delete moduleMap[platform];
+        if (Object.keys(moduleMap).length === 1) {
+          delete map[id];
+        }
+        let dupsByPl = hasteMap.duplicates[id];
+        if (dupsByPl == null) {
+          dupsByPl = hasteMap.duplicates[id] = (Object.create(null): any);
+        }
```

Comment:
You could turn this into const, which is imho more readable, but up to you:

```
const dupsByPl = hasteMap.duplicates[id] == null
  ? (hasteMap.duplicates[id] = (Object.create(null): any))
  : hasteMap.duplicates[id];
```

I guess it only matters if you are trying to avoid `let` :)